src/overview/components/DateRangeSelection.jsx

```diff
@@ -1,46 +1,126 @@
-import React, { PropTypes } from 'react'
+import React, { Component, PropTypes } from 'react'
 import moment from 'moment'
+import chrono from 'chrono-node'
 import DatePicker from 'react-datepicker'
 import 'react-datepicker/dist/react-datepicker-cssmodules.css'
 
 import styles from './DateRangeSelection.css'
 
 
-const DateRangeSelection = ({
-    startDate, endDate,
-    onStartDateChange, onEndDateChange,
-}) => (
-    <div className={styles.dateRangeSelection}>
-        <DatePicker
-            className={styles.datePicker}
-            dateFormat='DD-MM-YYYY'
-            placeholderText='after..'
-            isClearable
-            selected={startDate && moment(startDate)}
-            selectsStart
-            startDate={moment(startDate || 0)}
-            endDate={moment(endDate)}
-            maxDate={moment()}
-            onChange={date => onStartDateChange(
-                date ? date.valueOf() : undefined,
-            )}
-        />
-        <DatePicker
-            className={styles.datePicker}
-            dateFormat='DD-MM-YYYY'
-            placeholderText='before..'
-            isClearable
-            selected={endDate && moment(endDate)}
-            selectsEnd
-            startDate={moment(startDate || 0)}
-            endDate={moment(endDate)}
-            maxDate={moment()}
-            onChange={date => onEndDateChange(
-                date ? date.endOf('day').valueOf() : undefined,
-            )}
-        />
-    </div>
-)
+class DateRangeSelection extends Component {
+    constructor() {
+        super()
+
+        this.state = {
+            startDateText: null,
+            endDateText: null,
+        }
+
+        this.handleKeydown = this.handleKeydown.bind(this)
+        this.submitDateChange = this.submitDateChange.bind(this)
+    }
+
+    componentDidMount() {
+        this.refs.startDateDatePicker.refs.input.refs.input.addEventListener('keydown', e => this.handleKeydown(e, 'startDate'))
+        this.refs.endDateDatePicker.refs.input.refs.input.addEventListener('keydown', e => this.handleKeydown(e, 'endDate'))
```

Comment:
Yes these two lines are bad. Do you have any experience what would be the best thing to do. 

https://github.com/Hacker0x01/react-datepicker/blob/master/src/datepicker.jsx#L269 This function in datepicker is the one that's causing the problem. "react-datepicker" also wants to listen to key presses so there's no simple way (as far as I know) to prevent this "onInputKeyDown" function on datepicker from being called. Unless you get to the event before React and stop it from propagating. The worst thing is there's really a simple fix to solve this if there was a way to pass in onKeyDown as props to datepicker component.

Do you know of any other way to get this event before datepicker component? Since we're higher in the tree datepicker component always gets it first. Unless you attach a native listener to this input field what I did.

I can extract this to a separate component but I don't know how much the quality of the code will improve since those two lines would still exist and there would just be a lot of passing the same props.